Ehr 12345 Summary view table stylin, admin action buttons, and icon refactors

jsed500 opened the pull request.

sambiramairelogic commented on components/styled/icon/src/atoms/cross-icon.tsx:
```diff
@@ -1,18 +1,38 @@
 
-const CrossIcon: FC<ExternalLinkIconProps> = ({ size, className, ...rest }) => (
-  <FontAwesomeIcon
+const calculateIconColor = (status: string): SerializedStyles => css`
```
status could be a TS type instead of an open string. 

Linguistically I find it odd to read "status: 'red'", I'd argue warning is a status but red is a colour. I also realise that remember that warning === red and success === green is extra cognitive load. 

How about we expand this method to accept either "success" or "green" if you want green, etc.?

Moza805 commented on components/styled/icon/src/atoms/info-circle-icon.tsx:
```diff
@@ -6,11 +6,11 @@ import { faInfoCircle } from '@fortawesome/free-solid-svg-icons'
-  ${status === 'red' && `${ICON_COLOURS.DANGER};`}
-  ${status === 'default' && `${ICON_COLOURS.DEFAULT};`}
-  ${status === 'info' && `${ICON_COLOURS.INFO};`}
+  ${status === 'green' && `${ICON_COLOURS.SUCCESS.VALUE} !important;`}
```
Is there a nicer way to achieve this without using `!important` ? From experience this usually causes more issues than it solves

Moza805 commented on components/styled/icon/src/molecules/icon.tsx:
```diff
@@ -83,11 +83,11 @@ const Icon: FC<IconProps> = ({ type, size, status, direction, counterValue, anim
 
     case 'cross': {
-      return <CrossIcon size={size} {...rest} />
+      return <CrossIcon size={size} status={status ?? 'default'} {...rest} />
```
You could default this in the `CrossIcon` component to avoid having this wherever we don't want to set status

Moza805 commented on components/styled/table/src/molecules/table-cell.tsx:
```diff
@@ -26,7 +28,29 @@ const TableCell: FC<CellProps> = ({
+      <ActionMenu
+        actions={adminActions}
+        menuButtonOptions={{
+          type: 'button',
```
Sorta seems like a default `type` of `button` would be nice here

Moza805 commented on components/styled/button/src/atoms/button.tsx:
```diff
@@ -120,7 +131,8 @@ const StyledButton = styled.button<StyledProps>`
 
-  ${({ buttonStyle }): SerializedStyles => setColors(buttonStyle)}
+  ${({ padding }): SerializedStyles => setButtonPadding(padding)}
+  ${({ buttonStyle }): SerializedStyles => setColors(buttonStyle ?? 'primary')}
```
unnecessary duplication of the default between this and line 149, idm which you keep though

Moza805 commented on components/styled/button/src/atoms/button.tsx:
```diff
@@ -90,13 +90,24 @@ const setIconMargins = (placement: IconPlacement): SerializedStyles => {
+const setButtonPadding = (padding: string | undefined): SerializedStyles => {
+  if (padding) {
+    return css`
+      ${{ padding }};
```
should this be `padding: ${{ padding }};` ?

Moza805 commented on components/styled/button/src/atoms/button.tsx:
```diff
@@ -90,13 +90,24 @@ const setIconMargins = (placement: IconPlacement): SerializedStyles => {
   }
 }
 
+const setButtonPadding = (padding: string | undefined): SerializedStyles => {
```
Why would we use this rather than emotion e.g.

```javascript
const PaddingStyledButton = styled(Button)`
  padding: 1rem 2rem 3rem 4rem;
`
```

Moza805 reviewed (commented):
I see what @sambiramairelogic means about the table, I can't make head nor tail of what it's doing, would not like to have to maintain that 🙈 !

jsed500 commented:
I have now rewritten all icons using a single shared StyledIcon atom. Have deleted the individual implementations of icons as too much same code and too much copy paste. I have refactored all icons in the repo so now that all of them can be: Coloured, Spun (weee!!!!), transformed, and clickHandled as available - but not required - options. These colours can be overridden in usual emotion way.

sambiramairelogic commented on packages/styles/src/atoms/icons.ts:
```diff
@@ -6,50 +9,53 @@ type FontAwesomeIconSize = 'xs' | 'lg' | 'sm' | '1x' | '2x' | '3x' | '4x' | '5x'
 export type IconSize = 'small' | 'medium' | 'large'
 
-export type IconStatus = 'red' | 'green' | 'amber' | 'info' | 'default' | 'primary' | 'secondary'
+export type IconStatus = 'default' | 'red' | 'green' | 'amber' | 'info' | 'primary' | 'secondary' | 'link'
```
I'm still not a fan of this 'status' attribute - why not just call it colour? What's the concept of an icon's "status"?

sambiramairelogic commented on components/clinical/patient-banner/src/molecules/primary-info.tsx:
```diff
@@ -7,7 +7,7 @@ import {
   TABLET_MINIMUM_MEDIA_QUERY,
 } from '@ltht-react/styles'
-import { ChevronCircleIcon } from '@ltht-react/icon'
+import { Icon } from '@ltht-react/icon'
```
Shouldn't this be the default export?

sambiramairelogic commented on components/styled/icon/src/organisms/icon.tsx:
```diff
@@ -0,0 +1,337 @@
+  }
+}
+
+const calculateChevronIcon = (direction: IconDirection): IconDefinition => {
```
Looks like these two methods are identical between this file and icon-button.tsx ?

jsed500 replied:
Not sure where you are referring to...

sambiramairelogic reviewed (approved):
Love a good re-factor, thanks for taking the time and effort to simplify our icons code! 

(See I AM grateful, pwomis)
